mlrun/projects/project.py

```diff
@@ -2048,16 +2037,19 @@ def run(
             secrets=self._secrets,
             artifact_path=artifact_path,
             namespace=namespace,
+            source=source,
         )
-        run_msg = f"started run workflow {name} "
         # run is None when scheduling
-        if run:
-            run_msg += f"with run id = '{run.run_id}' "
-        run_msg += f"by {workflow_engine.engine} engine"
-        logger.info(run_msg)
+        if run and run.state == mlrun.run.RunStatuses.failed:
+            return run
+        if not workflow_spec.schedule:
+            # Failure and schedule messages already logged
+            logger.info(
+                f"started run workflow {name} with run id = '{run.run_id}' by {workflow_engine.engine} engine"
+            )
```

Comment:
you ask if `run` is not None, meaning it can be None yet you access `run.run_id` without validating it.

Reply:
`run=None` only when scheduling, so it is safe in here.
See the comment few lines above: "# run is None when scheduling"